### PR TITLE
Pgebheim/experimental connector

### PR DESCRIPTION
**Pull request by pgebheim**

### CODE REVIEW

**Comment by pgebheim** [comment]
@justinbarry -- Here is my first draft of this.

- It uses cross-fetch to bring in a cross-env `fetch` polyfill.
- It only supports HTTP at the moment
- The test is written inline 'cause I didn't spend the time to figure out how/where you're putting the tests for this stuff, this comes with a npm script called `connect` for @augurproject/sdk which just runs that one test. That should be refactored.

General thoughts:
- Augur SDK needs a pass to get the styles the same. We have multiple file name formats for classes etc.
- There are places where we are explicitly importing from `src` -- this will break as soon as this is used as an NPM module. Can't TS automatically point the module at `src` in TS-land so that when its an npm module the main `build` directory will be isomorphic to the old `src` dir and relative imports would still work?

**Comment by justinbarry** [comment]
1. Added discriminant union code to #2183.
2. Added code style issue #2207.

File: packages/augur-sdk/src/connector/http-connector.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Connector, Callback } from "./connector";
+// TODO: use cross-fetch?
```
Comment (justinbarry) [inline packages/augur-sdk/src/connector/http-connector.ts]:
```suggestion

```

Yes. Unless you prefer another package.

File: packages/augur-sdk/src/connector/http-connector.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import fetch from "cross-fetch";
+
+export class HTTPConnector extends Connector {
+
+  constructor(public readonly endpoint: string) {
+    super();
+  }
+
+  public async connect(params: any): Promise<any> {
+
```
Comment (justinbarry) [inline packages/augur-sdk/src/connector/http-connector.ts]:
```suggestion
return Promise.resolve();
```

File: packages/augur-sdk/src/connector/http-connector.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  public async disconnect(): Promise<any> {
+
```
Comment (justinbarry) [inline packages/augur-sdk/src/connector/http-connector.ts]:
```suggestion
return Promise.resolve();
```

**Review by justinbarry — CHANGES_REQUESTED** [review]
Requested simple changes. Otherwise looks good!

**Comment by pgebheim** [comment]
@brianosaurus -- Here is my solution to the `sender` thing.

**Comment by brianosaurus** [comment]
@justinbarry, if main in pakage.json points to src or build it'll automatically find the include. You and I chatted about a while ago. 

LGTM, I'll absorb this into the mostly finished websocket stuff. Personally I would have implemented BindTo in the base class and then just did comm out of the individual connectors. Pass it back and let API deal with deserializing  CC: @pgebheim

**Review by justinbarry — APPROVED** [review]

**Pull request merged**